Frontend/src/pages/waiting/components/WaitingList.jsx

```diff
@@ -0,0 +1,39 @@
+import { useQuery } from '@tanstack/react-query'
+import { useContext } from 'react'
+import { Table } from 'semantic-ui-react'
+import { CompetitionContext } from '../../../api/helper/context/competition_context'
+import { getWaitingCompetitors } from '../../../api/registration/get/get_waiting'
+import { setMessage } from '../../../ui/events/messages'
+import LoadingMessage from '../../../ui/messages/loadingMessage'
+
+export default function WaitingList() {
+  const { competitionInfo } = useContext(CompetitionContext)
+  const { isLoading, data: waiting } = useQuery({
+    queryKey: ['waiting', competitionInfo.id],
+    queryFn: () => getWaitingCompetitors(competitionInfo.id),
+    retry: false,
+    onError: (err) => {
+      setMessage(err.message, 'error')
+    },
+  })
+  return isLoading ? (
+    <LoadingMessage />
+  ) : (
```

Comment:
Won't this have that problem again of trying to display the waiting list in the error case when `waiting` is `null`?